zrcs/zrc-7.md

```diff
@@ -0,0 +1,216 @@
+| `resource_integrity` |      `String`      | A Base64 encoded SHA digest of the file pointed by the `resource`. This is an [integrity metadata](https://w3c.github.io/webappsec-subresource-integrity/#integrity-metadata-description). For example, if the SHA-256 is the hash function and `8z5D++W8NDHzFm5rY4/JxkXlIlU2cSQ65XjighJVk9U=` is the Base64 encoded SHA digest of the resource file, the `resource_integrity` is `sha256-8z5D++W8NDHzFm5rY4/JxkXlIlU2cSQ65XjighJVk9U=`. <br/><br/> If `resource` points to centralized storage, this property can be used to ensure the integrity of the resource. Otherwise, this property is unnecessary. |          |
+| `attributes`         | `Array of Objects` | An array of attributes.<br/><br/> Each attribute has the following properties: <ul> <li>`trait_type` : `String` <br/> The name of the trait. <br/> _(optional)_</li> <li>`value` : `String`, `Number`, or `Boolean` <br/> The value of the trait. When the value is `Number`, it should be integer or float. <br/> _(required)_</li> <li>`display_type` : `String` <br/> The display type of the trait. If the display type is `timestamp`, we recommend using unix timestamp. <br/> _(optional)_</li> </ul>                                                                                                 |          |
+| `external_url`       |      `String`      | A URL that points to an external website presenting the asset.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |          |
+| `animation_url`      |      `String`      | A URL to a multi-media attachment for the asset. The examples of file extensions are GLTF, GLB, WEBM, MP4, M4V, OGV, OGG, MP3, WAV, and OGA. <br/><br/> Also, `animation_url` can be HTML pages for interactive NFTs using JavaScript canvas, WebGL, etc.                                                                                                                                                                                                                                                                                                                                                    |          |
+
```

Comment:
I want to discuss adding `external_description` and `external_resource` to the specification. 

The main reason why is that currently, marketplaces require infomation to be passed to be displayed on the website either a textual description or some banner image. This adds a layer of friction which could have been fetched from metadata.

This data could be repeated across all tokens (just like external_url), but ecosystem partners should consume what's in token_id 1 json response for these two fields, allowing for reduced file size, reduced duplication and easier user-focused management of these fields across a collection.

Reply:
We will structure it as follows
1. Contract level metadata
2. Token level metadata

Contract level metadata can be used to contain things like descriptions for the entire NFT and is not token specific
Token level metadata will house individual token specific information such as attributes 

From a marketplace pov, it will just need to query the contract level metadata to get information about the nft. This will reduce the amount of duplication description for token level metadata